pkg/datapath/linux/config/config.go

```diff
@@ -356,13 +356,19 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		cDefinesMap["ENABLE_HOST_FIREWALL"] = "1"
 	}
 
-	if option.Config.EncryptInterface != "" {
-		link, err := netlink.LinkByName(option.Config.EncryptInterface)
+	if iface := option.Config.EncryptInterface; iface != "" {
+		link, err := netlink.LinkByName(iface)
```

Comment:
I was tempted to say we should throw an error here, but it I think its OK to try defaults if the link specified doesn't exist. At least we've been doing it this way so probably best to leave it.

Reply:
I know it's merged already, but to me it feels counter-intuitive to silently ignore the iface that the user suggested. Maybe add a warning if the link does not exist?